### ISSUE TITLE
Add support for TextField pk

### ISSUE THREAD

**Issue by arshsingh** [aaugustin/django-sesame#46]
I use a TextField pk, specifically https://github.com/ergeon/python-flax-id

**Comment by aaugustin** [comment]
Let's discuss adding support for `CharField`, which flax-id uses.

`TextField` is essentially "`CharField` so large that I don't want to bother specifying a max_length`. I expect a PK to have constant length so I'm not sure it makes sense to support `TextField` here. Let's wait for someone to actually need it.

Coming back to `CharField`, the question is "how to encode it in the sesame token".

Currently django-sesame is able to encode / decode integers and UUID, which are fixed length — 4 bytes and 16 bytes respectively.  The token format is designed to be as compact as possible (because long URLs are ugly, get truncated accidentally, etc.) So django-sesame just reads the expected number of bytes from the token. There's nothing else.

Since strings are variable length, the most generic format would be 1 byte for the length of the UTF-8 encoded string, followed by the UTF-8 encoded string. (Supporting only up to 255 characters would be an acceptable limitation for an user id that ends up in a URL.)

This wouldn't be optimal for flax-id, though:

- IDs are fixed length, so there's no need for the length byte
- It would be best to use IDs in the "12 bytes" form rather than in the "16 URL-safe characters" form because django-sesame always base64-encodes the whole token to make it URL-safe, and there's no point doing it twice

There's no way django-sesame can determine and handle all this, so the only way to allow these optimizations is to make the packer configurable.

----

To sum up:

1. I'd like to support custom packers in addition to the three natively supported packers — then you can do the optimal thing for flax-id if you want
2. I'm unsure about adding a `CharField` packer (and `BinaryField`, which would be the same, minus the UTF-8 encoding). As shown above, the generic implementation is unlikely to be the best implementation in most cases. Most users certainly care a lot more about having something that works and a lot less about having something optimal than I do...

**Comment by arshsingh** [comment]
This is great! Thank you :star2:
